### PR TITLE
Ensure menus closes when another one gets opened

### LINKED ISSUE #7936: Option on fields should collapse if another one is selected
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

I know this is a very minor and visual only issue which can probably even be ignored, but I decided to report it anyway.

While being in a collection, the 3 dots on the right open a menu (edit, half width etc...).
If another one is selected, I expect the already selected one to collapse and not stay open; instead it opens another popup behind the already existing one.

image

### To Reproduce

1. Create multiple fields in a collection
2. Select the menu on one, then select the menu on another one

### What version of Directus are you using?

9.0.0-rc.92

### What version of Node.js are you using?

v16.7.0

### What database are you using?

PostgreSQL 13.2

### What browser are you using?

Safari

### What operating system are you using?

macOS

### How are you deploying Directus?

Docker

### PR DESCRIPTION
**Pull request by azrikahar**
Fixes #7936

## Reported Bug

In the Data Model page, or basically anywhere with multiple menu buttons, other menu buttons will not close when we click on a new menu button:

fnfvLAETPc

## Solution

~~add `@focusout` event to trigger menu closing because menu will lose focus whenever we click/focus another menu.~~

reverted the above change since it's the wrong solution and causes other menus (especially nested menus) to stop working. Refer the discussions below for more info.

### Latest changes made:

- moved the click event to the label as suggested
- added `align-self: stretch` so the label hitbox will fill up the field-select vertically
- After turning `.label` to flex to center the name + interface text, the truncating stopped working because it doesn't play nice in flexboxes. Had to add a `.label-inner` and move the truncate text CSS properties into it.

### ~~Outdated result~~
uAy3lcyDSX

### Current result

Here we can see initially it shows the menu are closing properly when another one is opened. Later on the label is hovered to show it has the same coverage/hitbox as suggested in our discussion.

P5rvuwsMjb

Clearer indication of current hitbox:

chrome_dJ3zTUZb07

### CODE REVIEW

File: app/src/components/v-menu/v-menu.vue [hunk 1]
```diff
@@ -4,6 +4,7 @@
 			ref="activator"
 			class="v-menu-activator"
 			:class="{ attached }"
+			@focusout="deactivate"
```
Comment (rijkvanzanten) [inline app/src/components/v-menu/v-menu.vue:7]: How does this bubble up when you're in a nested menu? There's a handful of places where we have menus in menus (like the datetime or color interfaces). Does a child-focusout cause the parent to lose focus too?

**Comment by rijkvanzanten** [comment]
Also, the menus already have a "on click outside of the menu" handler that will deactivate the menu when you click on something else. However, I think that particular logic might rely on a class that's shared between the menus, instead of having a unique ID per menu, causing that handler to be ignored when you click on _another_ menu. Maybe that's the better fix here 🙂

File: app/src/components/v-menu/v-menu.vue (hunk 1 above)
Comment (azrikahar) [reply app/src/components/v-menu/v-menu.vue:7]: Oh right.. Thanks for mentioning the "menus in menus", didn't thought of that. Will try to look into it & see does it break or not (unfortunately most likely would from what I can think of).

**Comment by azrikahar** [comment]
> Also, the menus already have a "on click outside of the menu" handler that will deactivate the menu when you click on something else. However, I think that particular logic might rely on a class that's shared between the menus, instead of having a unique ID per menu, causing that handler to be ignored when you click on _another_ menu. Maybe that's the better fix here 🙂

Yea I did look into that, though I couldn't reach any conclusion in terms of solving the issue with it, but I'll look into it again 👍

Just as an extra info, originally I wanted to talk about the investigation I did, but ended up thinking it might be too much so I scratched that. 😅 

With that said, I originally started with thinking maybe we need to keep track of the open/close events and bubble it up to the parent containing the fields so we can close the others... though that definitely feels over-engineered. Then I did look into `v-click-outside` (the one you mentioned), thinking maybe we can do some handler magic there (basically the configs/handlers it offers) and can't seem to actually pinpoint why it's ignoring other menu (had the same thought process as you mentioned as well). So I ended up with the current fix when it suddenly pop up in my mind.

**Comment by rijkvanzanten** [comment]
Ah yeah to confirm, this is definitely causing some issues with nested dropdown, but also with cases where you have another element with focus nested, like an input. For example:


https://user-images.githubusercontent.com/9141017/132890319-e41a5129-461a-4b44-a00e-8d3c88075db0.mp4

**Comment by rijkvanzanten** [comment]
Found the issue!

The problem here is that the activator for this particular menu has a stop-propagation flag:

https://github.com/directus/directus/blob/e6c8660a81077af6a88183cd0463a85c3bf587db/app/src/modules/settings/routes/data-model/fields/components/field-select-menu.vue#L4

It's there to prevent the row click from happening (as that would navigate to open the field detail drawer. However, that also means it stops the propagation up the click event listener for the click-outside flag of the other menus. There's two things we can do to fix it: 

1) Use a different event (like pointerdown instead of click) for the click-outside directive, or;
2) Restructure the field-select boxes so only the name is clickable, instead of the whole box:

CleanShot 2021-09-10 at 13 12 27@2x

**Comment by azrikahar** [comment]
@rijkvanzanten Thanks for taking the time to investigate it and point me to the right direction! Once again I'm too fixated on v-menu and didn't even thought of looking for it in the field select and tackle it from there... 🤦‍♂️

Ended up doing the 2nd suggestion you gave since it feels like it makes more sense, at least to me. Will update the PR with the new changes now.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**